bundles/automation/org.eclipse.smarthome.automation.module.script/src/main/java/org/eclipse/smarthome/automation/module/script/internal/handler/AbstractScriptModuleHandler.java

```diff
@@ -90,23 +90,25 @@ public void dispose() {
     private void loadConfig() {
         Object type = module.getConfiguration().get(SCRIPT_TYPE);
         Object script = module.getConfiguration().get(SCRIPT);
-        if (type == null || !(type instanceof String) || ((String) type).trim().isEmpty()) {
-            throw new RuntimeException(
-                    String.format("Type is missing in the configuration of module '%s'.", module.getId()));
-        } else if (script == null || !(script instanceof String) || ((String) script).trim().isEmpty()) {
-            throw new RuntimeException(
-                    String.format("Script is missing in the configuration of module '%s'.", module.getId()));
+        if (isNotValid(type)) {
+            throw new IllegalStateException(String.format("Type is missing in the configuration of module '%s'.", module.getId()));
+        } else if(isNotValid(script)) {
+            throw new IllegalStateException(String.format("Script is missing in the configuration of module '%s'.", module.getId()));
         } else {
             this.type = (String) type;
             this.script = (String) script;
         }
     }
+    
+    private boolean isNotValid(Object parameter) {
```

Comment:
I see that we would like to check if the parameter is not valid in the cases above.
But I don't think that methods that check that something is not true should be used.
If we want to check if a collection is not empty, we call `!collection.isEmpty()` instead of `collection.isNotEmpty()`.
I would also prefer to provide a method `isValid(Object parameter)` and use `!isValid(type)` above.